lib/src/signed_video_h26x_sign.c

```diff
@@ -154,13 +154,13 @@ complete_sei_nalu_and_add_to_prepend(signed_video_t *self)
   return status;
 }
 
-/* Removes the first element from the SEI buffer of a `signed_video_t` structure,
+/* Removes the specified index element from the SEI buffer of a `signed_video_t` structure,
  * shifting remaining elements left and clearing the last slot. */
 static void
-shift_sei_buffer_index(signed_video_t *self)
+shift_sei_buffer_at_index(signed_video_t *self, int index)
 {
   const int sei_data_buffer_end = self->sei_data_buffer_idx;
-  for (int j = 0; j < sei_data_buffer_end - 1; j++) {
+  for (int j = index; j < sei_data_buffer_end; j++) {
```

Comment:
You lost the "-1". If it is not there you could potentially move in garbage into the buffer.